### ISSUE TITLE
Schema for Custom Payload

### ISSUE THREAD

**Issue by thejoefriel** [yalla-coop/death#116]
@ivanmauricio - do you think there's any more fields we should be adding?

Have looked to keep it very simple at this stage

```
{
    "botMsgs": [Array of strings -  messages from the bot],
    "questionType": String ("button", "freeformText", "textSkip" ),
    "options": [Array of strings - options for user to select]
}
```

Bit more context on the questionTypes:
- button: this is to render a series of buttons, each option when clicked would send that text as a user response to trigger the next intent
- freeformText: this is when the user can type as much as they want. Render a finished button which will trigger the next follow up intent (style looks the same as for store a memory etc)
- textSkip: this is text input from the user, but we also render a 'I'd rather not say' button. Clicking this would trigger the 'I'd rather not say' intent which always ends the existing flow and stores what they have entered at that point before selecting a new flor

**Comment by ivanmauricio** [comment]
Hey @thejoefriel, just have a few comments.

Does the `questionType` refer to the way the `options` are going to be displayed on the front-end? If so, will there ever be a case where an individual `option` will need to be displayed differently from the others. For the chatbot I worked on we had the options be an array of objects, so that metadata could be added to each individual one. I'm not sure if this is something you'll need but may be good to think about

```elm
const Options = ["Card", "Button"] // etc

{
    "botMsgs": [String],
    "options": [{type: Option, content: String }]
}
```

I'd also recommend having a defined list of the option types included in this schema. Obviously a first draft and you can add to them later but would be good for everyone to know where to go to look up the different types that can be used

**Comment by thejoefriel** [comment]
Thanks @ivanmauricio - no on this project we can render all the options the same I believe, but we'll stick with  the array of objects so this can be adapted easily in the future if it becomes more complex. 

Yeah the questionType refers to how they would be rendered and the logic (e.g. textSkip needs to render the text input and also a button that will say 'I'd rather not say' which will then have logic to end the flow)

So in summary then: 

```
{
    "botMsgs": [String],
    "questionType": String,
    "options": [{type: Option, content: String}]
}

const Options = [ "Button" ] 
const QuestionTypes = ["button", "freeformText", "textSkip"]
```

**Comment by ivanmauricio** [comment]
Yes that looks good! Let's check in when you have built things out a bit more, to see if this is working. As I said we should think of this as the first draft which we will extend if there's anything we haven't thought of. I'm not sure where would be a good place to put this, maybe in the readme?

**Comment by ivanmauricio** [comment]
Also small point but I guess it would be something more like 
```elm
{
    "botMsgs": [String],
    "questionType": QuestionType,
    "options": [{type: Option, content: String}]
}

const Options = [ "Button" ] 
const QuestionTypes = ["button", "freeformText", "textSkip"]
```

I just changed the type `String` for `questionType` field to be a type `QuestionType`. It is obviously still a `String` but this is us making a unique set of strings and saying that that field's value has to be one of them. Let me know if that makes sense!
